luarules/gadgets/game_unit_market.lua

```diff
@@ -99,17 +99,45 @@ local function setNotForSale(unitID)
     unitsForSale[unitID] = nil
 end
 
-local function setUnitOnSale(unitID, price, toggle)
-    if unitsForSale[unitID] == nil or unitsForSale[unitID] == 0 or toggle == false then
-        unitsForSale[unitID] = price
-        spSetUnitRulesParam(unitID, "unitPrice", price, RPAccess)
-        UnitSaleBroadcast(unitID, price, spGetUnitTeam(unitID))
-        setForSaleState(unitID, 1)
-        return true
-    else
+
+local t2conNames = {"armack", "armacv", "armaca", "armacsub", "corack", "coracv", "coraca", "coracsub", "legack", "legacv", "legaca", "legacsub"}
+local isT2Con = {}
+for _, name in ipairs(t2conNames) do
+    if UnitDefNames[name] then
+        isT2Con[UnitDefNames[name].id] = true
+    end
+end
```

Comment:
set t2conNames to nil after it has been used

Reply:
Is this by convention or for maybe garbage collection?

Reply:
it doesn't take a lot of memory, but it frees it up i believe

Reply:
Not really. The var isn't used anywhere else afterwards so the Lua interpreter knows on its own to free the memory.